src/loom-provider.ts

```diff
@@ -157,12 +158,21 @@ export class LoomProvider {
     this._ethRPCMethods = new Map<string, EthRPCMethod>()
     this.notificationCallbacks = new Array()
     this.accounts = new Map<string, Uint8Array>()
+    let privateKey: Uint8Array
 
     // Only subscribe for event emitter do not call subevents
     this._client.addListener(ClientEvent.EVMEvent, (msg: IChainEventArgs) =>
       this._onWebSocketMessage(msg)
     )
 
+    if (privateKeyOrAddress instanceof Uint8Array) {
+      privateKey = privateKeyOrAddress
+    } else {
+      privateKey = CryptoUtils.generatePrivateKey()
+      this.setMiddlewaresForAddress(privateKeyOrAddress.local.toString(), client.txMiddleware)
```

Comment:
This line I'm not sure about, the client middleware is usually setup to work with a specific DAppChain key, while the account address used here is most likely an Ethereum address (which requires different middleware). So reusing the client middleware here doesn't seem to make sense.

Reply:
Generally, the client is setup with whatever signer middleware we're using prior to the creation of the provider, at least in the dashboard and the examples. Unless there is some other use case?

Reply:
Hmm, I guess that sounds reasonable then.